connectors/src/connectors/microsoft/temporal/workflows.ts

```diff
@@ -1,29 +1,98 @@
 import type { ModelId } from "@dust-tt/types";
-import { proxyActivities } from "@temporalio/workflow";
+import {
+  continueAsNew,
+  executeChild,
+  proxyActivities,
+  workflowInfo,
+} from "@temporalio/workflow";
 
 import type * as activities from "@connectors/connectors/microsoft/temporal/activities";
 import type * as sync_status from "@connectors/lib/sync_status";
-import type { DataSourceConfig } from "@connectors/types/data_source_config";
+import type { MicrosoftNodeType } from "@connectors/resources/microsoft_resource";
 
-const { fullSyncActivity } = proxyActivities<typeof activities>({
+const { getSiteNodesToSync, syncFiles, markNodeAsVisited } = proxyActivities<
+  typeof activities
+>({
   startToCloseTimeout: "20 minutes",
 });
 
-const { syncSucceeded } = proxyActivities<typeof sync_status>({
+const { reportInitialSyncProgress, syncSucceeded } = proxyActivities<
+  typeof sync_status
+>({
   startToCloseTimeout: "10 minutes",
 });
 
 export async function fullSyncWorkflow({
   connectorId,
-  dataSourceConfig,
+  startSyncTs = undefined,
 }: {
   connectorId: ModelId;
-  dataSourceConfig: DataSourceConfig;
+  startSyncTs?: number;
 }) {
-  await fullSyncActivity({ connectorId, dataSourceConfig });
+  if (startSyncTs === undefined) {
+    startSyncTs = new Date().getTime();
+  }
+
+  await executeChild(fullSyncSitesWorkflow, {
+    workflowId: microsoftFullSyncSitesWorkflowId(connectorId),
+    searchAttributes: {
+      connectorId: [connectorId],
+    },
+    args: [{ connectorId, startSyncTs }],
+    memo: workflowInfo().memo,
+  });
+}
+
+export async function fullSyncSitesWorkflow({
+  connectorId,
+  startSyncTs,
+  nodesToSync = undefined,
+  totalCount = 0,
+}: {
+  connectorId: ModelId;
+  startSyncTs: number;
+  nodesToSync?: MicrosoftNodeType[];
```

Comment:
We should never pass models to workflows. Workflow arguments should be as minimal as possible. Any other example of a workflow taking models as argument?

(note: this will go to temporal, pricing of temporal is based on the state of workflows)

Reply:
I take it you mean model in the sense of "non-primitives types"? (not db models, that I'm not passing here ofc)

Ack, fixing

Reply:
yeah here a resource right same difference :)

Question: how big can this array be? are these selected nodes or all files?

Reply:
selected nodes, then child folders--but not files
similar to what is done in google drive

Reply:
Ok assuming this is very likely <= 100-200 that's fine but above that it's not awesome